circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/DropTableService.java

```diff
@@ -25,28 +28,69 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.hotels.bdp.circustrain.api.data.DataManipulationClient;
 import com.hotels.hcommon.hive.metastore.client.api.CloseableMetaStoreClient;
+import com.hotels.hcommon.hive.metastore.iterator.PartitionIterator;
 
 public class DropTableService {
 
   private static final Logger LOG = LoggerFactory.getLogger(DropTableService.class);
   private static final String EXTERNAL_KEY = "EXTERNAL";
   private static final String IS_EXTERNAL = "TRUE";
+  private static final short BATCH_SIZE = (short) 10;
 
   /**
    * Removes all parameters from a table before dropping the table.
    */
-  public void removeTableParamsAndDrop(
+  public void removeTableParamsAndDrop(CloseableMetaStoreClient client, String databaseName, String tableName)
+    throws TException {
+    Table table = getTable(client, databaseName, tableName);
+    if (table != null) {
+      dropTable(client, table, databaseName, tableName);
+    }
+  }
+
+  /**
+   * Drops the table and its associated data. If the table is unpartitioned the table location is used. If the table is
+   * partitioned then the data will be dropped from each partition location.
+   */
+  public void dropTableAndData(
       CloseableMetaStoreClient client,
       String databaseName,
-      String tableName) throws TException {
-    Table table;
+      String tableName,
+      DataManipulationClient dataManipulationClient)
+    throws TException {
+    LOG.debug("Dropping table {}.{} and its data.", databaseName, tableName);
+    Table table = getTable(client, databaseName, tableName);
+
+    if (table != null) {
+      String replicaLocation = table.getSd().getLocation();
+      if (table.getPartitionKeysSize() == 0) {
+        deleteData(dataManipulationClient, replicaLocation);
+      } else {
+        List<String> partitionLocations = getPartitionLocations(client, table);
```

Comment:
Are we OK with not doing the whole partition iterator trick here? I guess for most of this type of table they're generally not very big so it's unlikely this will return tens of thousands of partitions and run out of memory like we've had in other parts of the code so it's probably fine.

Reply:
Do you mean the hcommon PartitionIterator trick? b/c if so then it does use that in the `getPartitionLocations` method at the bottom of the class.
`PartitionIterator partitionIterator = new PartitionIterator(client, table, BATCH_SIZE);`